modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIEndpointRelevantObjectEntryModelListener.java

```diff
@@ -258,32 +295,51 @@ private void _validateSingleElementPath(
 				"x-must-start-with-the-x-character");
 		}
 
-		Matcher individualMatcher = _individualPathPattern.matcher(pathString);
+		Map<String, Serializable> values = objectEntry.getValues();
 
-		Matcher leftForwardMatcher = _leftSlashForwardPattern.matcher(
-			pathString);
+		if (Objects.equals(
+				values.get("scope"), HeadlessBuilderConstants.GROUP) &&
+			Objects.equals(
+				pathParameter, HeadlessBuilderConstants.PATH_PARAMETER_ID)) {
+
+			throw new ObjectEntryValuesException.InvalidObjectField(
+				Arrays.asList(objectField.getLabel(user.getLocale())),
+				"Single element ID endpoint can not be scoped by group",
+				"single-element-id-endpoint-can-not-be-scoped-by-group");
+		}
 
-		Matcher rightForwardMatcher = _rightSlashForwardPattern.matcher(
-			pathString);
+		String pathParameterString = StringUtil.extractLast(
+			pathString, StringPool.FORWARD_SLASH);
 
-		if ((leftForwardMatcher.matches() || rightForwardMatcher.matches()) &&
-			!individualMatcher.matches()) {
+		Matcher individualMatcher = _individualPathPattern.matcher(pathString);
 
+		if (!individualMatcher.matches()) {
 			throw new ObjectEntryValuesException.InvalidObjectField(
 				Arrays.asList(objectField.getLabel(user.getLocale())),
 				"%s can have a maximum of 255 alphanumeric characters",
 				"x-can-have-a-maximum-of-255-alphanumeric-characters");
 		}
+
+		Matcher curlyBraceMatcher = _curlyBracePattern.matcher(
+			pathParameterString);
+
+		if (!curlyBraceMatcher.matches()) {
+			throw new ObjectEntryValuesException.InvalidObjectField(
+				Arrays.asList(objectField.getLabel(user.getLocale())),
+				"%s must contain a path parameter between curly braces",
+				"x-must-contain-a-path-parameter-between-curly-braces");
+		}
 	}
 
+	private static final Pattern _curlyBracePattern = Pattern.compile(
+		"^\\{[a-zA-Z0-9]+\\}$");
 	private static final Pattern _individualPathPattern = Pattern.compile(
-		"/[a-zA-Z0-9][a-zA-Z0-9-/-{-}]{1,253}");
-	private static final Pattern _leftSlashForwardPattern = Pattern.compile(
-		".*[{].");
+		"/[a-zA-Z0-9][a-zA-Z0-9-/-{\\-}]{1,253}");
 	private static final Pattern _pathPattern = Pattern.compile(
 		"/[a-zA-Z0-9][a-zA-Z0-9-/]{1,253}");
-	private static final Pattern _rightSlashForwardPattern = Pattern.compile(
-		".*[}].");
+
+	@Reference
+	private APIApplicationProvider _apiApplicationProvider;
```

Comment:
I think this is not being used. Is this needed?

Reply:
make sense!! it is not needed